src/main/java/com/tradeshift/amqp/rabbit/properties/TunedRabbitProperties.java

```diff
@@ -1,234 +1,321 @@
 package com.tradeshift.amqp.rabbit.properties;
 
 import java.util.Objects;
-
-import org.springframework.boot.autoconfigure.amqp.RabbitProperties;
 import org.springframework.core.io.Resource;
+import org.springframework.util.StringUtils;
+
+public class TunedRabbitProperties {
```

Comment:
I am thinking about the way we are setting the default values.

You are setting the default values ​​using the `NullSafe.getOrDefault` method (which is a good strategy), but you are using the anemic model with this strategy (another class is responsible for changing the Entity values) and it is very difficult to find where are the default value.

My suggestion in this case is to use `NullSafe.getOrDefault` within` TunedRabbitProperties.get<Variable>`. With this, we have a rich model and if you want to find the default value just open the Entity.

What do you think?

Reply:
Hey, thanks for the review.

With the first solution, it was not possible to do something like you suggested,  but I made some changes and now it's possible with another way.

Reply:
Perfect solution! Thank you very much for your contribution.